res_syntax/cli/reactjs_jsx_v4.ml

```diff
@@ -817,6 +817,13 @@ let transformStructureItem ~config mapper item =
           React_jsx_common.raiseErrorMultipleReactComponent ~loc:pstr_loc
         else (
           config.hasReactComponent <- true;
+          let binding =
+            match binding.pvb_expr.pexp_desc with
+            | Pexp_record
+                ([({txt = Ldot (Ldot (Lident "Js", "Fn"), _)}, e)], None) ->
+              {binding with pvb_expr = e}
```

Comment:
Is it okay to lose the information whether it is an uncurried function? Don't we need to keep the `make` function uncurried?

Reply:
Good! I think we don’t need in most cases because v4 is having the record-based representation. Just in curious about the discouraged forward ref case.
Let me check little quick and will revisit if any issue.

Reply:
Found the parsing error in the case of the forwardRef component as uncurried. https://github.com/rescript-lang/rescript-compiler/issues/5807